### PR TITLE
Execution chapter merge

### PR DESCRIPTION
**Pull request by AcidSepp**
merged chapter-execution with chapter 3

### CODE REVIEW

File: LaTeX/Parts/Chapter3.tex [hunk 1]
```diff
@@ -1,8 +1,9 @@
 \chapter{Execution}\label{sec:Execution}
 To start a Varroa test the user needs to start a Commander and at least one Agent.
 The Agents periodically try to establish a connection to the Commander.
```
Comment (SgtSilvio) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
The Agents periodically tries to establish a connection to the Commander.
```

File: LaTeX/Parts/Chapter3.tex [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
-Once all Agents are connected the execution of the scenario starts.
-\textsl{}
+Once all Agents are connected the execution of the scenario starts.\\
+\\
+Varroa has two main modes of execution for now. One is to configure Varroa ahead of time manually or programmatically and then execute directly. The other is to use containers and utilize dynamic discovery.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
Varroa has two main modes of execution. One is to configure Varroa ahead of time manually or programmatically and then execute directly. The other is to use containers and utilize dynamic discovery.
```

File: LaTeX/Parts/Chapter3.tex [hunk 3]
```diff
@@ -37,8 +38,25 @@ \subsection{Agent}
 For better understanding of the temporal processes the just explained concepts are illustrated in figure \ref{pic:AgentStates}.
 
 \section{Configuration}
-\subsection{Commander}\label{sec:commanderConfig}
-\begin{lstlisting}[caption={Commander XML configuration}, captionpos=b, label={lst:commanderConfig}, language=XML]
+Following are the two sequence sketches for direct (static initialization) configuration \ref{pic:staticExecution} and dynamic (DNS discovery based) \ref{pic:dynamicExecution} configuration.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
The Following sketches are the two sequences for direct (static initialization) configuration \ref{pic:staticExecution} and dynamic (DNS discovery based) \ref{pic:dynamicExecution} configuration.
```

File: LaTeX/Parts/Chapter3.tex [hunk 4]
```diff
@@ -37,8 +38,25 @@ \subsection{Agent}
...
+\subsection{Static configuration}
+\begin{figure}[h]
+\begin{center}
+\includegraphics[scale=1]{Resources/PDF/ExecutionStaticInit}
+\caption{static execution}
+\label{pic:staticExecution}
+\end{center}
+\end{figure}
+
+Static configuration may be used on any infrastructure where hostname or IP addresses of all Varroa instances (Commander and Agents) are known ahead of time. This is the case on most typical systems, such as IaaS clouds where an instance is booted and assigned an address before the application is deployed onto that system.\\
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
A Static configuration may be used on any infrastructure where the hostnames or IP addresses of all Varroa instances (Commander and Agents) are known ahead of time. This is the case on most of the typically used systems, such as IaaS clouds where an instance is booted and is assigned an address before the application is deployed onto the system.\\
```

Comment (RobinAtherton) [reply LaTeX/Parts/Chapter3.tex]: Generally weird sentence structure. Someone should take a look over this.

File: LaTeX/Parts/Chapter3.tex [hunk 5]
```diff
@@ -37,8 +38,25 @@ \subsection{Agent}
...
+\\
+For static configuration, the user must provide an instance configuration, which configures the Varroa instance as a Commander or an Agent.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
To configure Varroa statically the user must provide an instance configuration, which sets up the Varroa instance as a Commander or an Agent.
```

File: LaTeX/Parts/Chapter3.tex [hunk 6]
```diff
@@ -37,8 +38,25 @@ \subsection{Agent}
...
+A detailed description of those configuration options can be found in the paragraphs below.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
A detailed description of these configuration options can be found in the paragraphs below.
```

**Review by RobinAtherton — CHANGES_REQUESTED** [review]

File: LaTeX/Parts/Chapter3.tex [hunk 7]
```diff
@@ -69,4 +87,43 @@ \subsection{Agent}\label{sec:agentConfig}
...
+
+\begin{figure}[h]
+\begin{center}
+\includegraphics[scale=1]{Resources/PDF/ExecutionDnsInit}
+\caption{dynamic execution}
+\label{pic:dynamicExecution}
+\end{center}
+\end{figure}
+
+DNS discovery specifically is aimed at container environments and cloud platforms. These types of platforms generally provide different methods of service discovery, which are necessary to allow the parts of a distributed systems to communicate with each other.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
DNS discovery is specifically aimed at container environments and cloud platforms. These types of platforms generally provide different methods of service discovery, which are necessary to allow the parts of a distributed system to communicate with each other.
```

File: LaTeX/Parts/Chapter3.tex [hunk 8]
```diff
@@ -69,4 +87,43 @@ \subsection{Agent}\label{sec:agentConfig}
...
+
+In container environments especially, the address of a server an application is going to run on is generally not known ahead of application startup. Containers are launched quickly and in an unpredictable order, leading to a scenario where the addresses of all parts of a system cannot be known ahead of time. This necessitates a configuration method which allows service discovery after application startup, which the previous method cannot handle.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
In container environments, the address of a server where an application is going to run on is generally not known ahead of application startup. Containers are launched quickly and in an unpredictable order, leading to a scenario where the addresses of all parts of a system are unknown ahead of time. This necessitates a configuration method which allows service discovery after application startup, which the previous method cannot handle.

```

File: LaTeX/Parts/Chapter3.tex [hunk 9]
```diff
@@ -69,4 +87,43 @@ \subsection{Agent}\label{sec:agentConfig}
...
+
+
+\begin{figure}[h]
+\begin{center}
+\includegraphics[scale=0.8]{Resources/PDF/ExecutionDnsDiscovery}
+\caption{agent record resolution}
+\end{center}
+\end{figure}
+
+At startup Varroa will continuously resolve those records, which must be a round-robin A record, and compare each address found in the records to the address corresponding to its own hostname. If any address matches its own, the instance will set its role according to the set of addresses it matches.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
During start up Varroa will continuously resolve these records, which must be designed as round-robins.
Then each address found in the records is compared to the address corresponding to its own hostname. 
If a match is found, the instance will set its role according to the set of addresses it matches.
```

File: LaTeX/Parts/Chapter3.tex [hunk 10]
```diff
@@ -69,4 +87,43 @@ \subsection{Agent}\label{sec:agentConfig}
...
+
+Additionally, agents can use the commander address to locate the commander node and continuously attempt to connect to it until it is ready for agent connections.
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
Additionally Agents can use the Commanders address to locate its node and continuously attempt to connect to it until it is ready for the Agents' connection requests.
```
Don't commit without looking at this, still unsure about these changes contentwise.

File: LaTeX/Parts/Chapter3.tex [hunk 11]
```diff
@@ -69,4 +87,43 @@ \subsection{Agent}\label{sec:agentConfig}
...
+
+The commander can also determine how many agents will connect to it using the agent address, or alternatively a agent count can be  provided at startup (for deployments on services where records are slowly updated as containers start).
```
Comment (RobinAtherton) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
The commander can also determine how many Agents will connect to it, using the Agents address, or alternatively an agent count can be  provided at startup (for deployments on services where records are slowly updated as containers start).
```

**Review by RobinAtherton — CHANGES_REQUESTED** [review]

File: LaTeX/Parts/Chapter3.tex (hunk 6 above)
Comment (AcidSepp) [reply LaTeX/Parts/Chapter3.tex]: just no

Comment (AcidSepp) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
A detailed description of these configuration options can be found in the paragraphs below.
```

File: LaTeX/Parts/Chapter3.tex (hunk 10 above)
Comment (AcidSepp) [reply LaTeX/Parts/Chapter3.tex]: Content is okay, but Agent and Commander should be should be capitalized.

File: LaTeX/Parts/Chapter3.tex (hunk 11 above)
Comment (AcidSepp) [reply LaTeX/Parts/Chapter3.tex]: Agent and Commander should be capitalized.

**Pull request merged**